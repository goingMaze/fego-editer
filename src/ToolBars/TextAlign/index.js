import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Modifier, EditorState } from 'draft-js'
import { getSelectedBlocksMetadata } from 'draftjs-utils'
import { ToolBarBtn } from '../../utils'
import Iconfont from '../../iconfont'

const TEXTALIGN_TYPES = [
	{ label: 'left', style: 'left', icon: 'iconLeftAlign', title: '左对齐' },
	{ label: 'center', style: 'center', icon: 'iconCenterAlign', title: '居中' },
	{ label: 'right', style: 'right', icon: 'iconRightAlign', title: '右对齐' },
	{ label: 'justify', style: 'justify', icon: 'iconJustify', title: '两端对齐' },
]

export default class TextAlign extends Component {
	static propTypes = {
		editorState: PropTypes.object.isRequired,
		onChange: PropTypes.func.isRequired,
		config: PropTypes.array.isRequired,
	}
	constructor(props) {
		super(props)
		this.state = {
			currentTextAlignment: undefined
		}
	}
	componentWillReceiveProps = nextProps => {
		if (nextProps.editorState !== this.props.editorState) {
			this.setState({
				currentTextAlignment: getSelectedBlocksMetadata(nextProps.editorState).get('text-align')
			})
		}
	}
	onToggle = type => {
		let { editorState, onChange } = this.props;
		let { currentTextAlignment } = this.state;
		let textAlignType = undefined;
		if (currentTextAlignment !== type) {
			textAlignType = type
		}
		let newContentState = Modifier.setBlockData(editorState.getCurrentContent(), editorState.getSelection(),
			{ 'text-align': textAlignType });
		onChange(EditorState.push(editorState, newContentState, 'change-block-data'))
	}
	render() {
		let { currentTextAlignment } = this.state;
		let { config } = this.props;
		return (
			<div className="FegoEditor-toolbars">
				{
					TEXTALIGN_TYPES.filter(item => config.includes(item.label)).map((type) =>
						<ToolBarBtn
							key={type.label} title={type.title}
							active={type.style === currentTextAlignment}
							onToggle={this.onToggle}
							style={type.style}
						>
							{Iconfont[type.icon]}
						</ToolBarBtn>
					)
				}
			</div>
		)
	}
}