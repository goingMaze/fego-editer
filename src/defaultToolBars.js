const defaultToolBars = {
	options: ['BlockType', 'Color', 'FontFamily', 'FontSize', 'Image', 'Inline', 'Link', 'List', 'TextAlign', 'Video', 'History'],
	BgColor: [],
	BlockType: ['正文', 'H1', 'H2', 'H3', 'H4', 'H5', 'H6'],
	Color: ['rgb(97,189,109)', 'rgb(26,188,156)', 'rgb(84,172,210)', 'rgb(44,130,201)',
		'rgb(147,101,184)', 'rgb(71,85,119)', 'rgb(204,204,204)', 'rgb(65,168,95)', 'rgb(0,168,133)',
		'rgb(61,142,185)', 'rgb(41,105,176)', 'rgb(85,57,130)', 'rgb(40,50,78)', 'rgb(0,0,0)',
		'rgb(247,218,100)', 'rgb(251,160,38)', 'rgb(235,107,86)', 'rgb(226,80,65)', 'rgb(163,143,132)',
		'rgb(239,239,239)', 'rgb(255,255,255)', 'rgb(250,197,28)', 'rgb(243,121,52)', 'rgb(184,49,47)',
		'rgb(209,72,65)', 'rgb(124,112,107)', 'rgb(209,213,216)'],
	FontFamily: [{
		name: 'SimHei', family: '"SimHei", "黑体"'
	}, {
		name: 'Yahei', family: '"Microsoft Yahei", "微软雅黑"'
	}, {
		name: 'KaiTi', family: '"KaiTi", "楷体"'
	}, {
		name: 'FangSong', family: '"FangSong", "仿宋"'
	}],
	FontSize: [16, 20, 30, 40, 50, 70, 100, 200],
	History: ['undo', 'redo'],
	Image: {
		uploadUrl: '',
		uploadCallback: null
	},
	Inline: ['bold', 'italic', 'underline'],
	Link: {},
	List: ['ul', 'ol', 'indent', 'outdent'],
	TextAlign: ['left', 'center', 'right', 'justify'],
	Video: {},
	Remove: {}
};

export default defaultToolBars
