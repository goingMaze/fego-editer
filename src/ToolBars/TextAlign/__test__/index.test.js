import React from 'react'
import { mount, render } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { EditorState, ContentState, convertFromHTML } from 'draft-js'
import TextAlign from '..'
import defaultToolBars from '../../../defaultToolBars'
import ModalManage from '../../../utils/ModalManage'

describe('TextAlign', () => {
	const contentState = ContentState.createFromBlockArray(convertFromHTML('<div>test</div>'))
	const editorState = EditorState.createWithContent(contentState)
	const wrapper = mount(
		<TextAlign editorState={editorState} onChange={() => { }}
			config={defaultToolBars.TextAlign} modalManage={new ModalManage()}
		/>
	);
	it('渲染时开头应该有一个div', () => {
		expect(wrapper.html().startsWith('<div')).toEqual(true)
	})
	it('should have a toolBarBtn child component', () => {
		expect(wrapper.find('ToolBarBtn').length).toBe(4)
	})
	it('快照', () => {
		const rendered = render(
			<TextAlign editorState={editorState} onChange={() => { }}
				config={defaultToolBars.TextAlign} modalManage={new ModalManage()}
			/>
		)
		expect(toJSON(rendered)).toMatchSnapshot()
	})
})