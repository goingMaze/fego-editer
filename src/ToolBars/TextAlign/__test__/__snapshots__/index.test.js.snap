// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAlign 快照 1`] = `
<div
  class="FegoEditor-toolbars"
>
  <div
    class="FegoEditor-toolBarBtn "
    title="左对齐"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M153.6 186.368v65.024h716.8V186.368H153.6z m456.192 195.072H153.6v65.024h456.192V381.44z m-456.192 261.12h716.8v-65.024H153.6v65.024z m0 195.072h456.192v-65.024H153.6v65.024z"
      />
    </svg>
  </div>
  <div
    class="FegoEditor-toolBarBtn "
    title="居中"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M128 224C128 206.326888 142.656038 192 160.022227 192L863.977773 192C881.663162 192 896 206.84375 896 224 896 241.673112 881.343962 256 863.977773 256L160.022227 256C142.336838 256 128 241.15625 128 224ZM320 416C320 398.326888 334.584352 384 351.424333 384L672.575667 384C689.930848 384 704 398.84375 704 416 704 433.673112 689.415648 448 672.575667 448L351.424333 448C334.069152 448 320 433.15625 320 416ZM128 608C128 590.326888 142.656038 576 160.022227 576L863.977773 576C881.663162 576 896 590.84375 896 608 896 625.673112 881.343962 640 863.977773 640L160.022227 640C142.336838 640 128 625.15625 128 608ZM320 800C320 782.326886 334.584352 768 351.424333 768L672.575667 768C689.930848 768 704 782.84375 704 800 704 817.673114 689.415648 832 672.575667 832L351.424333 832C334.069152 832 320 817.15625 320 800Z"
      />
    </svg>
  </div>
  <div
    class="FegoEditor-toolBarBtn "
    title="右对齐"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M153.6 186.368v65.024h716.8V186.368H153.6z m260.608 260.608H870.4V381.44H414.208v65.536zM153.6 642.56h716.8v-65.024H153.6v65.024z m260.608 195.072H870.4v-65.024H414.208v65.024z"
      />
    </svg>
  </div>
  <div
    class="FegoEditor-toolBarBtn "
    title="两端对齐"
  >
    <svg
      aria-hidden="true"
      class="icon"
      viewBox="0 0 1024 1024"
    >
      <path
        d="M153.6 186.368v65.024h716.8V186.368H153.6z m0 260.608h716.8V381.44H153.6v65.536z m0 195.584h716.8v-65.024H153.6v65.024z m0 195.072h716.8v-65.024H153.6v65.024z"
      />
    </svg>
  </div>
</div>
`;
